tools/cli/src/commands/task_queue.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{
+    args::{TargetEnvironment, TaskQueueArgs},
+    context::AppContext,
+};
+use anyhow::{Context, Result};
+use lavs_apis::id::TaskId;
+use lavs_task_queue::msg::{
+    ConfigResponse, CustomExecuteMsg, CustomQueryMsg, ExecuteMsg, QueryMsg, Requestor,
+};
+use layer_climb::{prelude::*, proto::abci::TxResponse};
+
+pub struct TaskQueue {
+    pub ctx: AppContext,
+    pub contract_addr: Address,
+}
+
+impl TaskQueue {
+    pub async fn new(ctx: AppContext, task_queue_args: &TaskQueueArgs) -> Result<Self> {
+        let addr_string = match task_queue_args.address.clone() {
+            Some(x) => x,
+            None => match ctx.args.target_env {
+                TargetEnvironment::Local => std::env::var("LOCAL_TASK_QUEUE_ADDRESS")
+                    .context("LOCAL_TASK_QUEUE_ADDRESS not found")?,
+                TargetEnvironment::Testnet => std::env::var("TEST_TASK_QUEUE_ADDRESS")
+                    .context("TEST_TASK_QUEUE_ADDRESS not found")?,
+            },
+        };
+
+        let contract_addr = ctx.chain_config.parse_address(&addr_string)?;
+
+        Ok(Self { ctx, contract_addr })
+    }
+
+    pub async fn add_task(
+        &self,
+        body: String,
+        description: String,
+        timeout: Option<u64>,
+    ) -> Result<(TaskId, TxResponse)> {
+        let payload = serde_json::from_str(&body).context("Failed to parse body into JSON")?;
+
+        let contract_config: ConfigResponse = self
```

Comment:
Okay, in this case the actual creator matters. Or can matter.
Some implementations of task queues gate on the calling address.

Maybe we don't use pools everywhere? (Good for deploy, but not the interactions with exisiting contracts that might associate state with the caller)

Reply:
for sure - if that's an explicit requirement we should make it explicit. Part of it is not using the pool (though, technically, it _is_ guaranteed to be derivation index `0` here, I don't like relying on that) - but also, even without a pool, if I'm understanding correctly you might want to have one wallet for deployment, but different one for adding tasks.

at the end of the day, should we have a separate env mnemonic for this - e.g. LOCAL_TASKADMIN_MNEMONIC?

Reply:
I would keep the same mnemonic. But use the primary (index 0) for tasks manipulating existing contracts (maybe take index as an explicit argument)? I love the pool for deployment

Maybe using something like `pool.get_index(0)` to specify what you want, and wait for it to be available if not (or panic?)

Reply:
opened an issue on the climb repo for that API (which would be great!): https://github.com/Lay3rLabs/climb/issues/11

will make it explicitly use that locally here for now...

Reply:
done in https://github.com/Lay3rLabs/avs-toolkit/pull/29/commits/d16b5c2188317eb2cdfb498ba15921c56f875d21